### DISCUSSION TITLE
How to get number of edge changes between this tree and the next?

### DISCUSSION THREAD

**Discussion by hyanwong** [tskit-dev/tskit#1581] [Q&A]
Given a tree, is there a way to find the number of edge changes to the next (or previous) tree? I can use the `.next()` method to get to the next one, but I can't see a way to find the number of changes involved. If there isn't one, I presume that I simply get the entire array by using `[len(ed.edges_in) for ed in ts.edge_diffs()]`?

**Comment by benjeffery** [comment]
`[len(ed.edges_in) for ed in ts.edge_diffs()]` only gets you the new edges, if you wanted number of changes don't you want `len(ed.edges_in)+len(ed.edges_out)`?

**Reply by hyanwong** [reply]
Ah, good point. I'm not sure if deleting one edge and then grafting the subtree somewhere else is a single "change" or two changes?

**Reply by minesrebollo** [reply]
So just to check I got this right, if I want to see how many edges changed from `ts.at_index(1000)` to` ts.at_index(1001)` I would do `[len(ed.edges_in)+len(ed.edges_out) for ed in ts.edge_diffs()][1001]` ?

**Comment by jeromekelleher** [comment]
There's no way of knowing this in the Tree API (the information isn't stored by the low-level code). The total number of in-and-out edges from the diffs provides a reasonable, easy, definition. If you start reasoning about what the edges do you'll get into all sorts of semantic awkwardnesses (esp. in cases where multiple recombinations happen).

**Reply by hyanwong** [reply]
Thanks Jerome. When you estimated that the number of edges changed between trees in the UKBB TS was ~40 (I think), was this the in+out numbers then? I assume that means that in an infinite-recombination-site TS, we should expect the number of edge "changes" between trees to be e.g. 6 (rather than 3 as we tend to quote)

**Reply by jeromekelleher** [reply]
I can't remember tbh, I had a look in the [repo](https://github.com/mcveanlab/treeseq-inference) but couldn't find where I did it. I guess you could also just do ``ts.num_edges / ts.num_trees`` as the number of edges per tree
